R/ui.R

```diff
@@ -128,6 +133,19 @@ ui_warn <- function(x, .envir = parent.frame()) {
   warning(x, call. = FALSE, immediate. = TRUE)
 }
 
+
+# Silence -----------------------------------------------------------------
+
+#' @rdname ui
+#' @param code Code to execute with usually UI output silenced.
+#' @export
+ui_silence <- function(code) {
+  old <- options(usethis.quiet = TRUE)
+  on.exit(options(old))
+
+  code
+}
+
```

Comment:
So this is basically `with_silence()`. Do we also want `local_silence()`? 🤔

Reply:
Also, FWIW we import withr if you wanted to use `withr::with_options()`

Reply:
Would you prefer `with_usethis_silence()` and `local_usethis_silence()`?

Reply:
My comment wasn't about the name as much as having both the "with" and "local" variants, since the need for one suggests you probably also need the other (or will soon).

Reply:
Yeah, but what we call it? 

We currently use `withr::local_options(list(usethis.quiet = TRUE))` in two places in the source; and I think `ui_silence()` would be better for both of them. We'd use a local silence more if the tests didn't silence by default.